### PR TITLE
update nginx configs

### LINKED ISSUE #443: Improve nginx error messages to not send back HTML
<hr/>
<details><summary>edit.</summary>
just felt I should mention that plain text or even just the numeric value from the HTTP status are probably the safest way to acknowledge errors. 
and just in-case you are considering something like a dialog alternative to alert (or even YT own polymer's dialogs) to render raw HTML from your server, please don't.

rendering external raw HTML is pretty much `eval`, 
and potentially can be used in-case the servers would be compromised to "drink up"
a lot of private information from the client-side.
</details>

### PR DESCRIPTION
**Pull request by mchangrh**
- handle CORS within nginx
- remove allow-headers from CORS
- add custom dynamic error pages
- remove unused fastcgi
- removed OPTIONS backend
- remove previous custom errorDirective
- moved proxy directives to appropiate file

closes #443

### CODE REVIEW

**Comment by mchangrh** [comment]
closes https://github.com/ajayyy/SponsorBlock/issues/1090

File: src/middleware/cors.ts [hunk 1]
```diff
@@ -2,7 +2,6 @@ import { NextFunction, Request, Response } from "express";
 
 export function corsMiddleware(req: Request, res: Response, next: NextFunction): void {
     res.header("Access-Control-Allow-Origin", "*");
-    res.header("Access-Control-Allow-Headers", "X-Requested-With, Content-Type, Accept");
```
Comment (ajayyy) [inline src/middleware/cors.ts:5]: Will this break cors when developing without nginx?

Comment (mchangrh) [reply src/middleware/cors.ts:5]: tests seemed to pass - Content-Type and Accept are whitelisted headers and X-Requested-With shouldn't be used.

I can write tests to check for PREFLIGHT

Comment (mchangrh) [reply src/middleware/cors.ts:5]: you're right content-type has to be explicitly whitelisted

**Comment by mchangrh** [comment]
see: #245 

accept restrictions shouldn't be bypassed and not sure what uses x-requested-with

**Pull request merged**